app/User.php

```diff
@@ -66,4 +66,9 @@ public function hasConnectedFacebook()
         return $this->facebook_connected;
     }
 
+    public function hasAlreadyConnectedWith($target_user)
+    {
+        return count(Connection::where('owner_id', $this->id)->where('user_id', $target_user->id)->get()) > 0;
```

Comment:
No need to execute a full `->get()` and wrapping in a count.

`return Connection::where('owner_id', $this->id)->where('user_id', $target_user->id)->count() > 0;`

Reply:
i actually refactored this myslf before i even read this